library/x509_crt.c

```diff
@@ -955,6 +956,10 @@ static int x509_get_crt_ext( unsigned char **p,
 
         if( ret != 0 )
         {
+            /* Give the callback (if any) a chance to handle the extension */
+            if (cb && cb(crt, &extn_oid, is_critical, p, end_ext_octet) == 0)
+                continue;
```

Comment:
Missing: check that `*p == end_ext_octet` on success. Otherwise the callback did something wrong.

Alternatively, I'm not sure that this check is needed here. Should the callback have to set `*p` to the end of the value (as currently documented)? Or should it do the check, and then `p` would be an input-only parameter (changed from `unsigned char **` to `const unsigned char *`)?

Reply:
Implemented in 4c3cb7a18d7347bb61426a79c46501256594f1cb